src/components/style-root.js

```diff
@@ -1,68 +1,53 @@
 /* @flow */
 
-import React, {PureComponent} from 'react';
+import React, {useContext, useRef} from 'react';
 import type {Node} from 'react';
-import PropTypes from 'prop-types';
 
 import Enhancer from '../enhancer';
 import StyleKeeper from '../style-keeper';
 import StyleSheet from './style-sheet';
 import type {Config} from '../config';
+import {StyleKeeperContext, RadiumConfigContext} from '../context';
 
-function _getStyleKeeper(instance): StyleKeeper {
-  if (!instance._radiumStyleKeeper) {
-    const userAgent =
-      (instance.props.radiumConfig && instance.props.radiumConfig.userAgent) ||
-      (instance.context._radiumConfig &&
-        instance.context._radiumConfig.userAgent);
-    instance._radiumStyleKeeper = new StyleKeeper(userAgent);
-  }
+function getStyleKeeper(
+  configProp: Config,
+  configContext?: Config
+): StyleKeeper {
+  const userAgent =
+    (configProp && configProp.userAgent) ||
+    (configContext && configContext.userAgent);
 
-  return instance._radiumStyleKeeper;
+  return new StyleKeeper(userAgent);
 }
 
 type StyleRootProps = {
   radiumConfig: Config,
   children: Node
 };
 
-class StyleRoot extends PureComponent<StyleRootProps> {
-  constructor() {
-    super(...arguments);
-
-    _getStyleKeeper(this);
-  }
-
-  getChildContext() {
-    return {_radiumStyleKeeper: _getStyleKeeper(this)};
-  }
-
-  _radiumStyleKeeper: StyleKeeper;
-
-  render() {
-    /* eslint-disable no-unused-vars */
-    // Pass down all props except config to the rendered div.
-    const {radiumConfig, ...otherProps} = this.props;
-    /* eslint-enable no-unused-vars */
-
-    return (
-      <div {...otherProps}>
-        {this.props.children}
-        <StyleSheet />
-      </div>
-    );
-  }
-}
-
-StyleRoot.contextTypes = {
-  _radiumConfig: PropTypes.object,
-  _radiumStyleKeeper: PropTypes.instanceOf(StyleKeeper)
+const StyleRootInner = Enhancer(({children, ...otherProps}: StyleRootProps) => (
+  <div {...otherProps}>
+    {children}
+    <StyleSheet />
+  </div>
+));
+
+const StyleRoot = (props: StyleRootProps) => {
+  /* eslint-disable no-unused-vars */
+  // Pass down all props except config to the rendered div.
+  /* eslint-enable no-unused-vars */
+  const {radiumConfig} = props;
+
+  const configContext = useContext(RadiumConfigContext);
+  const styleKeeper = useRef<StyleKeeper>(
```

Comment:
this won't change if `radiumConfig` prop changes -- but that is true to the original implementation